src/v1/components/surfaces/FloatingPanel/FloatingPanelProvider.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { createContext, useState } from "react";
+
+interface PanelState {
+  visible: boolean;
+  minimised: boolean;
+}
+
+interface PanelsType extends Record<string, PanelState> {}
```

Comment:
Suggestion: `[pr-author-resolves]` **Non-blocking** Nothing gives me more satisfaction that enumerating each specific keys
```suggestion
interface PanelsType extends Record<'floating' | 'normal\>, PanelState> {}
```

Reply:
The key could be anything. What is your intention behind having them as floating or normal? The current use case would be like 
```
{
  map1: {},
  timeline: {},
  map2: {}
....
}
```